docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md

```diff
@@ -9,25 +9,247 @@ menu:
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
+
+- `SET TRANSACTION READ ONLY` applies only to the current transaction block.
+- `SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY` applies the read-only setting to all statements and transaction blocks that follow.
+- `SET default_transaction_read_only = TRUE` applies the read-only setting to all statements and transaction blocks that follow.
+- `/*+ Set(transaction_read_only true) */ SELECT` applies only to the current transaction/select statement.
+
+## Examples
+
+This example uses follower reads since the **transaction** is marked read-only.
+
+```sql
+set yb_read_from_followers = true;
+start transaction read only;
+SELECT * from t WHERE k='k1';
+commit;
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+```
+
+This example uses follower reads since the **session** is marked read only.
+
+```sql
+set session characteristics as transaction read only;
+set yb_read_from_followers = true;
+SELECT * from t WHERE k='k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+The following examples use follower reads since **pg_hint** can be used during PREPARE, CREATE FUNCTION, and SELECT to do follower reads.
+
+{{< note title="Note" >}}
+The pg_hint needs to be applied at the prepare/function-definition stage and not at the `execute` stage.
+{{< /note >}}
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+PREPARE select_stmt(text) AS
+/*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k=$1;
+
+EXECUTE select_stmt(‘k1’);
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+CREATE FUNCTION func() RETURNS text AS
+$$ /*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k=1 $$ LANGUAGE SQL;
+CREATE FUNCTION
+SELECT func();
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+/*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k='k1';
+```
+
+```output
+----+----
+ k1 | v1
+(1 row)
+```
+
+A **join** example that uses follower reads.
+
+```sql
+create table table1(k int primary key, v int);
+create table table2(k int primary key, v int);
+insert into table1 values (1, 2), (2, 4), (3, 6), (4,8);
+insert into table2 values (1, 3), (2, 6), (3, 9), (4,12);
+set yb_read_from_followers = true;
+set session characteristics as transaction read only;
+select * from table1, table2 where table1.k = 3 and table2.v = table3.v;
+```
+
+```output
+ k | v | k | v
+---+---+---+---
+ 3 | 6 | 2 | 6
+(1 row)
+```
+
+The following examples demonstrate **staleness** after enabling the `yb_follower_read_staleness_ms` property.
+
+```sql
+set session characteristics as transaction read write;
```

Comment:
@amitanandaiyer , @aishwarya24 :

between 'v1' becoming 'v1+1' don't you need a 10sec sleep to illustrate the point. I don't see that being explicitly mentioned. Would be good to show the INSERT of 'v1' and a sleep 10s after that also, in my opinion.

Reply:
So the assumption here was that the initial state of the row was `v1` (as depicted by the select). With the write being made (at least 5+ seconds) in the past.

we could add it here, to make it more explicitly, if the default assumption is that the table/row is empty.

So replace the

```
insert into t values ('k1', 'v1')
/* sleep 10s */
```